### PR TITLE
Enable MP-JWT subsystem when not enabled

### PR DESCRIPTION
**Pull request by honza-kasik**
This PR enables MP-JWT subsystem and adds an extension if not already added for each test.

FYI @mchoma 

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Link to the passing job is provided https://eap-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/eap-7.x-microprofile-testsuite/205/
- [x] Code is self-descriptive and/or documented

### CODE REVIEW

**Comment by mchoma** [comment]
I have looked how FT and OpenApi do the same thing. And all 3 modules do in different way. 
- FT use classic `@Before/AfterClass`
- OpenApi use `MicroProfileServerSetupTask`

You are proposing ClassRule approach. Did you considered one of existing approaches above?

**Comment by honza-kasik** [comment]
Yes I did, `@ClassRule` seemed as the most elegant solution in this case

> FT use classic @Before/AfterClass

This does exactly the same thing with less code redundancy (two lines vs four).

> OpenApi use MicroProfileServerSetupTask

The same thing as above - too much boilerplate code IMHO.

**Comment by mchoma** [comment]
I agree that all approaches do the same. When SetupTask class is pulled into own class, than ClassRule and Setup Task approach use same number of lines in test class. Right?

Actually that is how it is done in `MultipleJarDeploymentTest` (fourth way).

**Comment by mchoma** [comment]
@honza-kasik do you agree with changing to SetupTask?

**Comment by honza-kasik** [comment]
@mchoma Sorry, notification lost in mail. Yeah, sure, I can switch it to `SetupTask`. Thanks!

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/EnableJwtSubsystemSetupTask.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    private boolean wasExtensionAdded;
+    private boolean wasSubsystemAdded;
+
+    @Override
+    public void setup() throws IOException, OperationException, TimeoutException, InterruptedException,
+            ConfigurationException {
+        try (final OnlineManagementClient client = ManagementClientProvider.onlineStandalone()) {
+            final Operations ops = new Operations(client);
+            /*
+             * if (!ops.exists(MP_JWT_EXTENSION_ADDRESS)) {
```
Comment (mchoma) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/EnableJwtSubsystemSetupTask.java:32]: @honza-kasik I thought we want to add extension if it is not there by default? Also `FaultToleranceServerSetup` do so.

Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/EnableJwtSubsystemSetupTask.java]: Thanks, this is already addressed in commit I forgot to push, so I'll have to restart the job.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/EnableJwtSubsystemRule.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    private boolean wasExtensionAdded;
+    private boolean wasSubsystemAdded;
+
+    @Override
+    protected void before() throws IOException, OperationException, TimeoutException, InterruptedException,
+            ConfigurationException {
+        try (final OnlineManagementClient client = ManagementClientProvider.onlineStandalone()) {
+            final Operations ops = new Operations(client);
+            /*
```
[the remark this answers is not in the document]
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/EnableJwtSubsystemRule.java]: +1

**Comment by honza-kasik** [comment]
@mchoma Added link to passing job, fixed description, removed WIP, ready to be reviewed.

**Comment by mchoma** [comment]
I think it is good to be merged. I am not sure if it is me who should merge. @mnovak1 ?

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/EnableJwtSubsystemSetupTask.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ * A way enable/disable MP-JWT subsystem when not enabled
+ */
+public class EnableJwtSubsystemSetupTask implements MicroProfileServerSetupTask {
+
+    private static final Address MP_JWT_EXTENSION_ADDRESS = Address
+            .extension("org.wildfly.extension.microprofile.jwt-smallrye");
+    private static final Address MP_JWT_ADDRESS = Address.subsystem("microprofile-jwt-smallrye");
+
+    private boolean wasExtensionAdded;
+    private boolean wasSubsystemAdded;
```
Comment (mnovak1) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/EnableJwtSubsystemSetupTask.java:25]: Optional: For non-java native programmers might be difficult to know what is default value for `boolean` thus I prefer to initialize it explicitly to `false`.

**Comment by mnovak1** [comment]
@honza-kasik just one minor thing to consider but LGTM to merge

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/EnableJwtSubsystemSetupTask.java (hunk 3 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/EnableJwtSubsystemSetupTask.java:25]: Thanks, will update now.

**Comment by honza-kasik** [comment]
@mnovak1 Thanks! Default value for boolean vars added. Also, the reload when adding subsystem is now done only when it was added.

**Comment by mnovak1** [comment]
@honza-kasik thanks, merged

**Pull request merged**